src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -441,11 +452,39 @@ BlockIO InterpreterDropQuery::executeToDatabaseImpl(const ASTDropQuery & query,
     return {};
 }
 
+void InterpreterDropQuery::extendQueryLogElemImpl(DB::QueryLogElement & elem, const DB::ASTPtr & ast, DB::ContextPtr context_) const
+{
+    auto & drop = ast->as<ASTDropQuery &>();
+    if (drop.database_and_tables)
+    {
+        auto & list = drop.database_and_tables->as<ASTExpressionList &>();
+        for (auto & child : list.children)
+        {
+            auto identifier = dynamic_pointer_cast<ASTTableIdentifier>(child);
+            if (!identifier)
+                throw Exception(ErrorCodes::SYNTAX_ERROR, "Unexpected type for list of table names.");
+
+            String query_database = identifier->getDatabaseName();
+            String query_table = identifier->getTableName();
+            if (!query_database.empty() && query_table.empty())
+            {
+                elem.query_databases.insert(backQuoteIfNeed(query_database));
```

Comment:
Seems only the last elem is logged.

Another question: Why we implement it in this PR? Did any test fail because of it?

Reply:
> Seems only the last elem is logged.
> 
Don’t quite understand what this sentence means.

> Another question: Why we implement it in this PR? Did any test fail because of it?
>
Yes, some tests failed. The reason is that the dabase name and table name are saved in `database_and_tables` in the new syntax, resulting in the database name and table name not being collected in `system.query_log`.